pkg/pillar/types/errortime.go

```diff
@@ -74,6 +162,26 @@ func (etsPtr *ErrorAndTimeWithSource) SetError(errStr string, errTime time.Time)
 	etsPtr.Error = errStr
 	etsPtr.ErrorSourceType = ""
 	etsPtr.ErrorTime = errTime
+	etsPtr.ErrorSeverity = ErrorSeverityError
+	etsPtr.ErrorEntities = []*ErrorEntity{}
+	etsPtr.ErrorRetryCondition = ""
+}
+
+// SetErrorWithDescription - Sets error state with ErrorDescription and no source type
+// it sets ErrorSeverityError in case of unspecified ErrorSeverity
+// it sets ErrorTime to time.Now() in case of no time provided
+func (etsPtr *ErrorAndTimeWithSource) SetErrorWithDescription(errDescription ErrorDescription) {
+	if errDescription.Error == "" {
```

Comment:
Ditto

Reply:
modified

Reply:
But can't we have a common function as part of ErrorDescription and have that common function be used by both ErrorAndTime and ErrorAndTimeWithSource?